include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_dispatchers.h

```diff
@@ -216,9 +216,15 @@ __onesweep_impl(sycl::queue __q, _RngPack1&& __input_pack, _RngPack2&& __virt_pa
         __esimd_radix_sort_onesweep_histogram<_KernelName>>;
     using _EsimdRadixSortScan = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
         __esimd_radix_sort_onesweep_scan<_KernelName>>;
-    using _EsimdRadixSortSweep =
-        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<::std::conditional_t<
-            __has_values, __esimd_radix_sort_onesweep_by_key<_KernelName>, __esimd_radix_sort_onesweep<_KernelName>>>;
+    using _EsimdRadixSortSweepInitial = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
+        ::std::conditional_t<__has_values, __esimd_radix_sort_onesweep_by_key<std::decay_t<_RngPack1>, std::decay_t<_RngPack2>, _KernelName>,
+                             __esimd_radix_sort_onesweep<std::decay_t<_RngPack1>, std::decay_t<_RngPack2>, _KernelName>>>;
```

Comment:
I'm not strictly opposed of that approach, but there is a drawback: kernel names will become longer (I guess, much longer).

The motivation of the change is to separate kernel names for the potential arrangements of `_RngPack1`, `_RngPack2`, `_RngPack3`, isn't it? If so, what do you think about using some constant instead, e.g.:

- ` __esimd_radix_sort_onesweep<0, ...>` instead of `__esimd_radix_sort_onesweep<std::decay_t<_RngPack1>, std::decay_t<_RngPack2>, ...>`,  
- `__esimd_radix_sort_onesweep<1, ...>` instead of `__esimd_radix_sort_onesweep<std::decay_t<_RngPack3>, std::decay_t<_RngPack2>, ...>`?

Reply:
The goal is to have a unique kernel name if and only if we have a unique kernel.  We want to avoid same mangled name error, but also we don't want to add extra kernel compilation only to support a differently named kernel when the types match exactly.  

Lets say `_RngPack1` is the exact same as `_RngPack3`, we then want `__esimd_radix_sort_onesweep<std::decay_t<_RngPack1>, std::decay_t<_RngPack2>, ...>`
and `__esimd_radix_sort_onesweep<std::decay_t<_RngPack3>, std::decay_t<_RngPack2>, ...>` to be the exact same as well.  If not, I think we require an extra kernel compilation only to satisfy having a different name.

If all three exactly match, we only want one kernel and one kernel name.

Reply:
That's reasonable, thanks for describing. 

Probably, it is still possible to use shorter kernel names. I will think about it separately as it is rather a "nice to have" thing.

Currently, I see the following options:

Rng1 type | Rng2 type | Rng3 type | Kernels | Constants
-- | -- | -- | -- | --
A | B | C | A->B, B->C, C->B | 0,1, 2
A | A | A | A->A, A->A, A->A | 3, 3, 3
A | A | B | A->A, A->B, B->A | 3, 0, 4
A | B | A | A->B, B->A, A->B | 0, 4, 0
B | A | A | B->A, A->A, A->A | 4, 3, 3

Reply:
I'm interested in the idea, but to create such constants, you would need to basically check `is_same<decay_t<_RngPack1>, decay_t<_RngPack2>>` for all three combos and assign constants appropriately.  
The code is ugly either way.  Perhaps the shorter kernel name is worth it.